c2corg_ui/static/js/documentediting.js

```diff
@@ -0,0 +1,269 @@
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.http_ = $http;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.apiUrl_ = apiUrl;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.module_ = $attrs['appDocumentEditing'];
+  goog.asserts.assert(goog.isDef(this.module_));
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.modelName_ = $attrs['appDocumentEditingModel'];
+
+  /**
+   * @type {?number}
+   * @private
+   */
+  this.id_ = null;
+
+  /**
+   * @type {?string}
+   * @private
+   */
+  this.culture_ = null;
+
+  if ('appDocumentEditingId' in $attrs &&
+      'appDocumentEditingCulture' in $attrs) {
+    this.id_ = $attrs['appDocumentEditingId'];
+    this.culture_ = $attrs['appDocumentEditingCulture'];
+    // Get document attributes from the API to feed the model:
+    this.http_.get(this.buildUrl_('read')).then(
+        goog.bind(this.successRead_, this),
+        goog.bind(this.errorRead_, this)
+    );
+  }
+};
+
+
+/**
+ * @const
+ * @type {string}
+ */
+app.DocumentEditingController.VIEW_PROJ = 'EPSG:4326';
+
+
+/**
+ * @const
+ * @type {string}
+ */
+app.DocumentEditingController.DATA_PROJ = 'EPSG:3857';
+
+
+/**
+ * @param {string} type Type of URL.
+ * @return {string} URL.
+ * @private
+ */
+app.DocumentEditingController.prototype.buildUrl_ = function(type) {
+  var url = this.apiUrl_;
+  url += url.substr(-1) != '/' ? '/' : '';
+  url += this.module_;
+  switch (type) {
+    case 'read':
+      url += '/' + this.id_ + '?l=' + this.culture_;
+      break;
+    case 'update':
+      url += '/' + this.id_;
+  }
+  return url;
+};
+
+
+/**
+ * @param {Object} response Response from the API server.
+ * @private
+ */
+app.DocumentEditingController.prototype.successRead_ = function(response) {
+  var data = response['data'];
+  if ('geometry' in data && data['geometry']) {
+    var geometry = data['geometry'];
+    if ('geom' in geometry && geometry['geom']) {
+      var geojson = new ol.format.GeoJSON();
+      var point = /** @type {ol.geom.Point} */
+          (geojson.readGeometry(geometry['geom']));
+      point.transform(app.DocumentEditingController.DATA_PROJ,
+                      app.DocumentEditingController.VIEW_PROJ);
+      var coordinates = point.getCoordinates();
+      data['geometry']['geom'] = {
+        'longitude': coordinates[0],
+        'latitude': coordinates[1]
+      };
+    }
+  }
+  this.scope_[this.modelName_] = data;
+};
+
+
+/**
+ * @param {Object} response Response from the API server.
+ * @private
+ */
+app.DocumentEditingController.prototype.errorRead_ = function(response) {
+  // TODO: better error handling
+  alert('Failed loading the document: ' +
+      response.status + ' ' + response.statusText);
+};
+
+
+/**
+ * @param {boolean} isValid True if form is valid.
+ * @export
+ */
+app.DocumentEditingController.prototype.submitForm = function(isValid) {
+  if (!isValid) {
```

Comment:
Maybe it's too early, but using the classes `ng-dirty` and `ng-valid`/`ng-invalid` you can easily indicate if a field is valid/invalid. Something like:

    .ng-dirty.ng-valid {
      border-color: green;
    }
    .ng-dirty.ng-invalid {
      border-color: red;
    }

https://docs.angularjs.org/api/ng/directive/form